rust/src/sdp/logger.rs

```diff
@@ -0,0 +1,154 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// written by Giuseppe Longo <giuseppe@glongo.it>
+
+use crate::jsonbuilder::{JsonBuilder, JsonError};
+
+use super::parser::{ConnectionData, MediaDescription, SdpMessage};
+
+pub fn sdp_log(msg: &SdpMessage, js: &mut JsonBuilder) -> Result<(), JsonError> {
+    js.open_object("sdp")?;
+
+    let origin = format!(
+        "{} {} {} {} {} {}",
+        &msg.origin.username,
+        &msg.origin.sess_id,
+        &msg.origin.sess_version,
+        &msg.origin.nettype,
+        &msg.origin.addrtype,
+        &msg.origin.unicast_address
+    );
+
+    js.set_string("origin", &origin)?;
+    js.set_string("session_name", &msg.session_name)?;
+
+    if let Some(session_info) = &msg.session_info {
+        js.set_string("session_info", session_info)?;
+    }
+    if let Some(uri) = &msg.uri {
+        js.set_string("uri", uri)?;
+    }
+    if let Some(email) = &msg.email {
+        js.set_string("email", email)?;
+    }
+    if let Some(phone_number) = &msg.phone_number {
+        js.set_string("phone_number", phone_number)?;
+    }
+    if let Some(conn_data) = &msg.connection_data {
+        log_connection_data(conn_data, js)?;
+    }
+    if let Some(bws) = &msg.bandwidths {
+        log_bandwidth(bws, js)?;
+    }
+    js.set_string("time", &msg.time)?;
+    if let Some(repeat_time) = &msg.time_zone {
```

Comment:
s/msg.time_zone/msg.repeat_time/

Reply:
also, an SV test? :)

Reply:
you mean an SV test for this specific field?

Reply:
yeah that would have caught this

Reply:
Would it be possible to have one that includes all (or most) of the fields we see in the logger?